### PR TITLE
feat: add statistics and /statistics/transactions endpoint

### PR DESCRIPTION
**Pull request by sborrazas**
refs #1492

### CODE REVIEW

File: lib/ae_mdw/stats.ex [hunk 1]
```diff
@@ -209,6 +223,87 @@ defmodule AeMdw.Stats do
...
+      {{:transactions, tx_tag}, :day, Util.min_int()},
+      {{:transactions, tx_tag}, :day, Util.max_int()}
+    }
+
+    fn direction ->
+      Collection.stream(state, Model.Statistic, direction, key_boundary, cursor)
+    end
+  end
+
+  defp render_statistic(state, {{:transactions, _tx_tag}, :day, interval_start} = statistic_key) do
```
Comment (jyeshe) [inline lib/ae_mdw/stats.ex:263]: For me it's okay if you wanna add `:month` and `:year` to the same PR, but it's your choice.

File: lib/ae_mdw/db/sync/stats.ex [hunk 2]
```diff
@@ -51,6 +59,44 @@ defmodule AeMdw.Db.Sync.Stats do
         &(&1 - 1)
       )
 
+  @spec statistics_mutations(time(), type_counts()) :: StatisticsMutation.t() | nil
+  def statistics_mutations(time, type_counts) do
```
Comment (jyeshe) [inline lib/ae_mdw/db/sync/stats.ex:63]: may we have a unit test for this syncing function and rename it for the purpose of the transactions stats?

File: lib/ae_mdw/stats.ex [hunk 3]
```diff
@@ -30,9 +34,17 @@ defmodule AeMdw.Stats do
...
   @typep aexn_type() :: :aex9 | :aex141
+  @typep query() :: map()
+  @typep pagination() :: Collection.direction_limit()
+  @typep pagination_cursor() :: Collection.pagination_cursor()
 
   @type nft_stats :: %{nfts_amount: non_neg_integer(), nft_owners: non_neg_integer()}
   @typep template_id() :: AeMdw.Aex141.template_id()
+
+  @type statistic_tag() :: {:transactions, Node.tx_type() | :all}
+  @type interval() :: :day | :week | :month
```
Comment (jyeshe) [inline lib/ae_mdw/stats.ex:45]: `interval_by` would be more appropriate

**Review by jyeshe — COMMENTED** [review]
Looking good, minor suggests and missed only sync tests to avoid regression.

File: lib/ae_mdw/stats.ex (hunk 3 above)
Comment (sborrazas) [reply lib/ae_mdw/stats.ex]: Done

File: lib/ae_mdw/db/sync/stats.ex (hunk 2 above)
Comment (sborrazas) [reply lib/ae_mdw/db/sync/stats.ex:63]: Done

File: lib/ae_mdw/stats.ex (hunk 1 above)
Comment (sborrazas) [reply lib/ae_mdw/stats.ex:263]: Yeah, was intending not to implement the filter yet. Also, monthly intervals would need to be rendered differently

**Review by jyeshe — APPROVED** [review]

**Pull request merged**